### ISSUE TITLE
data problem

### ISSUE THREAD

**Issue by starkcn** [yassouali/pytorch_segmentation#6]
could you show the more  data directory messages ? just like what files in the data directory ,the list of train data.

**Comment by yassouali** [comment]
Hi,

Are you talking generally or for a specific dataset,
If you want to add you dataset, just create a new loader in  `dataloader/`, the new class inherits from base_loader and you only need to implement `_set_files` (setting the paths of the images and the labels) and `_load_data` (where you load a specific pair of image / label depending on the argument passes), the data directory is up to you, check the examples of the implemented loaders.

For one of the four datasets, in read me I've detailed the files you'll need, if you have any more questions about a specific dataset, I can detail its files.

**Comment by starkcn** [comment]
Hello, I want to use your model to train my own dataset, I found that your way of loading data is somewhat different from the way other models I saw previously. If I want to train my own dataset, do I need to provide two folders, train and val, there is an image file and a label file in both folders, the image file contains the original image, the label  file contains the label images (stained image).Do I still need to provide a file similar to the one shown below? 

i dont know how to make my own txt just like this ?or,make my new txt like voc?
------------------ 原始邮件 ------------------
发件人: "Yassine"<notifications@github.com>;
发送时间: 2019年8月28日(星期三) 凌晨5:12
收件人: "yassouali/pytorch_segmentation"<pytorch_segmentation@noreply.github.com>;
抄送: "1501537826"<1501537826@qq.com>;"Author"<author@noreply.github.com>;
主题: Re: [yassouali/pytorch_segmentation] data problem (#6)




Hi,
 
Are you talking generally or for a specific dataset,
 If you want to add you dataset, just create a new loader in  dataloader/, the new class inherits from base_loader and you only need to implement _set_files (setting the paths of the images and the labels) and _load_data (where you load a specific pair of image / label depending on the argument passes), the data directory is up to you, check the examples of the implemented loaders.
 
For one of the four datasets, in read me I've detailed the files you'll need, if you have any more questions about a specific dataset, I can detail its files.

**Comment by yassouali** [comment]
It is up to you how you load you dataset, for creating a loader you have  two ways of doing things, creating a loader  that takes an argument split (val and train), and then you create two dataloaders for each, this is the way the provided datasets are implemented. Or you can only create one dataloader and then pass  the argument `val_split` and this way, you'll get a train loader and a also the val loader (calling `get_val_loader`). Take a look into `base_loader.py` and `base_datasets.py` and one of the example loaders to see how this is implemented.

All in all, first is to create your dataloader, by implementing two functions `_set_files` and `_load_data`

**Comment by starkcn** [comment]
i see，but i dont know the role of voc.txt、voc_context_classes-400.txt、voc_context_classes-59.txt.

**Comment by yassouali** [comment]
Oh, you can just ignore them, they're not used, they only map the class indices to the class name for each dataset, can be somewhat useful when debugging or you'd like to know the class predicted.

**Comment by starkcn** [comment]
your mean is,if i use this model to test my data,it will show me the label represented by the color.

**Comment by yassouali** [comment]
Not really no, those files are not used at all, for the color palettes, their defined in `utils/palette.py`.

**Comment by starkcn** [comment]
i see,thanks a lot!

**Comment by soans1994** [comment]
@yassouali 

sorry for reopening the issue, but i searched the closed issues regarding the custom dataloader. I am creating custom dataloader like your voc loader implementation. i have the raw image and segment maps in png. The train, train_val, val, test texts must be created? what is the difference between train_val and val?

thank you
